packages/auto-label/src/auto-label.ts

```diff
@@ -0,0 +1,256 @@
+};
+
+handler.checkExistingIssueLabels = async function checkExistingIssueLabels(
+  github: GitHubAPI,
+  owner: string,
+  repo: string,
+  issueNumber: number
+) {
+  try {
+    const data = await github.issues.listLabelsOnIssue({
+      owner,
+      repo,
+      issue_number: issueNumber,
+    });
+    if (data.data === undefined) {
+      return null;
+    } else {
+      return data.data;
+    }
+  } catch (err) {
+    return null;
+  }
+};
+
+handler.callStorage = async function callStorage(
+  bucketName: string,
+  srcFileName: string,
+  destFileName: string
+) {
+  const { Storage } = require('@google-cloud/storage');
+  const storage = new Storage();
+  const destFilename = path.resolve(destFileName);
+
+  const options = {
+    // The path to which the file should be downloaded, e.g. "./file.txt"
+    destination: destFilename,
+  };
+
+  // Downloads the file
+  await storage
+    .bucket(bucketName)
+    .file(srcFileName)
+    .download(options);
+
+  console.log(
+    `gs://${bucketName}/${srcFileName} downloaded to ${destFilename}.`
+  );
+  return path.resolve(__dirname, `../../${destFileName}`);
+};
+
+handler.checkIfFileIsEmpty = async function checkIfFileIsEmpty(
+  jsonData: string
+) {
+  if (fs.readFileSync(jsonData).length === 0) {
+    console.log('JSON file downloaded from Cloud Storage was empty');
+    return null;
+  } else {
+    const jsonArray = JSON.parse(fs.readFileSync(jsonData)).repos;
+    return jsonArray;
+  }
+};
+
+handler.checkIfElementIsInArray = function checkIfElementIsInArray(
+  jsonArray: JSONData[],
+  owner: string,
+  repo: string
+) {
+  if (jsonArray) {
+    const objectInJsonArray = jsonArray.find(
+      (element: JSONData) => element.repo === owner + '/' + repo
+    );
+    return objectInJsonArray;
+  } else {
+    return null;
+  }
+};
+
+function handler(app: Application) {
+  app.on(['issues.opened', 'issues.reopened'], async context => {
+    const owner = context.payload.repository.owner.login;
+    const repo = context.payload.repository.name;
+    const issueId = context.payload.issue.number;
+
+    const jsonData = await handler.callStorage(
+      'devrel-dev-settings',
+      'public_repos.json',
+      'src/downloadedfile.txt'
+    );
+
+    const jsonArray = await handler.checkIfFileIsEmpty(jsonData);
+
+    let objectInJsonArray: JSONData | null | undefined;
+    objectInJsonArray = handler.checkIfElementIsInArray(jsonArray, owner, repo);
+
+    if (objectInJsonArray == null || objectInJsonArray === undefined) {
+      console.log('There was no match for the repo name: ' + repo);
+      return;
+    }
+
+    const colorNumber = jsonArray.findIndex(
+      (object: JSONData) => objectInJsonArray === object
+    );
+    const githubLabel = objectInJsonArray.github_label;
+    const alreadyExists = await handler.checkExistingLabels(
+      context.github,
+      owner,
+      repo,
+      githubLabel
+    );
+
+    if (alreadyExists == null || alreadyExists === undefined) {
```

Comment:
I'd probably make this `===`, `null`, I try to avoid using `==` because JavaScript is too loose with it, it would work for `null`, `undefined`, false, 0, ''.